apps/prairielearn/src/pages/studentDetail/studentDetail.html.tsx

```diff
@@ -0,0 +1,187 @@
+import { Fragment } from 'preact/jsx-runtime';
+import { z } from 'zod';
+
+import { FriendlyDate } from '../../components/FriendlyDate.js';
+import { Scorebar } from '../../components/Scorebar.js';
+import { setCookieClient } from '../../lib/client/cookie.js';
+import {
+  StaffCourseInstanceSchema,
+  StaffEnrollmentSchema,
+  StaffUserSchema,
+} from '../../lib/client/safe-db-types.js';
+import { type StaffGradebookRow, computeLabel, computeTitle } from '../../lib/gradebook.shared.js';
+
+export const UserDetailSchema = z.object({
+  user: StaffUserSchema,
+  course_instance: StaffCourseInstanceSchema,
+  enrollment: StaffEnrollmentSchema.nullable(),
+  role: z.string(),
+});
+
+type UserDetail = z.infer<typeof UserDetailSchema>;
+
+interface StudentDetailProps {
+  gradebookRows: StaffGradebookRow[];
+  student: UserDetail;
+  urlPrefix: string;
+  courseInstanceUrl: string;
+}
+
+export function StudentDetail({
+  gradebookRows,
+  student,
+  urlPrefix,
+  courseInstanceUrl,
+}: StudentDetailProps) {
+  const { user, course_instance, enrollment, role } = student;
+
+  const gradebookRowsBySet = new Map<string, StaffGradebookRow[]>();
+  gradebookRows.forEach((row) => {
+    const setHeading = row.assessment_set.heading;
+    if (!gradebookRowsBySet.has(setHeading)) {
+      gradebookRowsBySet.set(setHeading, []);
+    }
+    const setAssessments = gradebookRowsBySet.get(setHeading);
+    if (setAssessments) {
+      setAssessments.push(row);
+    }
+  });
+
+  const handleViewAsStudent = () => {
+    setCookieClient(['pl_requested_uid', 'pl2_requested_uid'], user.uid);
+    setCookieClient(['pl_requested_data_changed', 'pl2_requested_data_changed'], 'true');
+    window.location.href = `${courseInstanceUrl}/assessments`;
+  };
+
+  const handleViewGradebookAsStudent = () => {
+    setCookieClient(['pl_requested_uid', 'pl2_requested_uid'], user.uid);
+    setCookieClient(['pl_requested_data_changed', 'pl2_requested_data_changed'], 'true');
+    window.location.href = `${courseInstanceUrl}/gradebook`;
+  };
+
+  return (
+    <div class="container-fluid">
+      <div class="card mb-4">
+        <div class="card-header bg-primary text-white d-flex align-items-center justify-content-between">
+          <h1 class="mb-0">Details</h1>
+          <button type="button" class="btn btn-sm btn-light" onClick={handleViewAsStudent}>
+            <i class="fas fa-user-graduate me-1" aria-hidden="true" />
+            View as Student
```

Comment:
We generally use `Sentence casing` on buttons like these. 
```suggestion
            View as student
```